vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java

```diff
@@ -978,8 +971,7 @@ private boolean processWays(QueryParameters queryParameters, ITileDataSink mapDa
                     GeoPoint[][] wayNodesArray = wayNodes.toArray(new GeoPoint[wayNodes.size()][]);
                     if (!filterRequired || !wayFilterEnabled || wayFilterBbox.intersectsArea(wayNodesArray)) {
                         List<Tag> tags = new ArrayList<>();
-                        for (int i = 0; i < e.tags.numTags; i++)
-                            tags.add(e.tags.tags[i]);
+                        tags.addAll(Arrays.asList(e.tags.getTags()));
```

Comment:
Should use the iteration over the actual tags size, like done with POIs above:
```java
for (int i = 0; i < e.tags.size(); i++)
    tags.add(e.tags.get(i));
```

Reply:
Oh yes! I adopted this from an older path, where i reseted the whole list and null all values. Thanks!